tests/functional/u/unnecessary/unnecessary_lambda_assignment.py

```diff
@@ -0,0 +1,29 @@
+"""Test assignment of lambda expressions to a variable."""
```

Comment:
Could you add the following test case:
```python
DICT = {1: lambda x: x, 2: lambda x: x + 1}

for key, value in DICT.items():
    print(value(key))
```

Similar patterns are reasons why I use the `lambda x: x` pattern and I want to make sure we don't accidentally regress and flag this in the future.

Reply:
Yeah this is fine, I've added it in to the unit test 😄